exp/api/v1beta1/ipam_types.go

```diff
@@ -0,0 +1,118 @@
+/*
+Copyright 2021 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package v1beta1
+
+import (
+	corev1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+)
+
+const (
+	// IPPoolLabel is used to add the pool name to an IPAddressClaim and IPAddress.
+	IPPoolLabel = "ipam.cluster.x-k8s.io/pool"
```

Comment:
It is, but this way you can easily filter by pool. Primarily useful for the in-cluster provider, but we should have it in general imo.

Reply:
Just feels a bit duplicate to have that info as a field and as a label. 

Is it about kubectl? If it's about CR client we can just export a func for an index

Reply:
An index also works, yeah.
I'd tend towards keeping it for kubectl anyway, but I'm also fine with removing it.

Reply:
Just feels not ideal to store the information twice.

Definitely open to keep it. Although I would prefer hearing other opinions.

Reply:
Below my comments on this topic:
- Due to historical reason, we exp folder contains MP experimental feature, but every other feature we added subsequently are in an exp subfolder. Should this go in exp/ipam in order to be consistent with the recent wave of experimental features? Also, if this goes into a different folder, our package folder will be cleaner and it opens up to have a different API group...
- Machine Pool was originally implemented in a different API group, but then we did a breaking change to bring them into CAPI main API group, which is where we expect them to lend when the experiment moves to GA. given this precedent:
  - I would like to avoid another breaking change for users, so IPAM types should start in the API group we expect them to land at the end of graduation
  - Given that ipam is a well-scoped feature that someone can use or not, I don't see a counter-argument for having this separated from the core API, e.g. in the ipam.cluster.x-k8s.io group starting with v1alpha1 version, but I think this decision should be approved in the office hours.

Reply:
@fabriziopandini Could it be that this comment belongs to my other comment below? (https://github.com/kubernetes-sigs/cluster-api/pull/6313#discussion_r912722893)

Reply:
WRT to the label I don't have strong opinions.
I usually prefer to add labels only if they are required by the code or there is a clear use case/demand for them (instead of adding them upfront because it could be useful); also in this case the information already surfaces as a column in kubectl get.
But I don't want to block on this and if we feel that it could avoid providers to introduce ad-hoc solutions, I'm ok with it

Reply:
Tendency is towards removing, so I'll do that. We can re-add it later if necessary.